test/unit/math/prim/mat/prob/vector_rng_test_helper.hpp

```diff
@@ -0,0 +1,364 @@
+#include <stan/math/prim/mat.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <test/unit/math/prim/scal/prob/util.hpp>
+
+enum class Constraint { None, Positive, NonNegative };
+
+/*
+ * Resize v to be length N and fill v with values that satisfy the given constraint c.
+ *
+ * @tparam T1 Type of v
+ * @param v Vector to fill with values
+ * @param c Constraint that values must fulfill
+ * @param N Number of elements that v will be resized to
+ */
+template <typename T1>
+void generateValues(T1& v, Constraint c, int N) {
```

Comment:
It seems handwavy to have the test values be so arbitrary (-1, 0, 1, 2, etc). I could make it fancier, but I think this does the job fine.